app/src/main/java/ch/epfl/sdp/appart/map/helper/MapFrontendHelper.java

```diff
@@ -0,0 +1,104 @@
+package ch.epfl.sdp.appart.map.helper;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import ch.epfl.sdp.appart.ad.Ad;
+import ch.epfl.sdp.appart.database.DatabaseService;
+import ch.epfl.sdp.appart.location.Location;
+import ch.epfl.sdp.appart.location.geocoding.GeocodingService;
+import ch.epfl.sdp.appart.location.place.Place;
+import ch.epfl.sdp.appart.location.place.address.AddressFactory;
+import ch.epfl.sdp.appart.location.place.locality.LocalityFactory;
+import ch.epfl.sdp.appart.map.MapService;
+import ch.epfl.sdp.appart.scrolling.card.Card;
+
+public class MapFrontendHelper {
+
+    private static final Pattern extractPostalCodePattern =
+            Pattern.compile("\\d+");
+    /**
+     * This corresponds to the maximum distance from the current user
+     * position at which apartments get drawn on the map.
+     */
+    private static final Float MAX_DISTANCE = 50_000.0f;
+
+    public static CompletableFuture<List<Card>> retrieveCards(DatabaseService databaseService) {
+        CompletableFuture<List<Card>> futureCards =
+                databaseService.getCards();
+        futureCards.exceptionally(e -> {
+            e.printStackTrace();
+            return null;
+        });
+        return futureCards;
+    }
+
+    public static void centerOnCurrentLocation(MapService mapService,
+                                               Location currentLocation) {
+        if (currentLocation != null) {
+            mapService.zoomOnPosition(currentLocation,
```

Comment:
Can `mapService` be `null`? If yes we should handle the case

Reply:
No, because this is called from the MapActivity which gets the service from Hilt.